apps/prairielearn/src/lib/authz-data.ts

```diff
@@ -1,156 +1,214 @@
+import assert from 'assert';
+
 import * as sqldb from '@prairielearn/postgres';
 import { run } from '@prairielearn/run';
 
-import { FullAuthzDataSchema } from './authz-data-lib.js';
-import { type User } from './db-types.js';
+import { type CalculateAuthDataResult, FullAuthzDataSchema } from './authz-data-lib.js';
+import {
+  type EnumCourseInstanceRole,
+  type EnumCourseRole,
+  type EnumMode,
+  type User,
+} from './db-types.js';
 
 const sql = sqldb.loadSqlEquiv(import.meta.url);
 
-export async function selectAuthzData({
+/**
+ * If course_id is not provided, but course_instance_id is,
+ * the function will use the course_id from the course instance.
```

Comment:
Nit for pettier Intellisense hover:

```suggestion
 * If `course_id` is not provided, but `course_instance_id` is,
 * the function will use the `course_id` from the course instance.
```